File: runtime/examples/android/VadJni/app/src/main/java/com/baidu/paddlespeech/vadjni/MainActivity.java
package com.baidu.paddlespeech.vadjni;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import com.baidu.paddlespeech.vadjni.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'vadjni' library on application startup.
    static {
        System.loadLibrary("vadjni");
    }

    private ActivityMainBinding binding;
    private long instance;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Example of a call to a native method
        TextView tv = binding.sampleText;
        tv.setText(stringFromJNI());

        Button lw = binding.loadWav;
    }

    /**
     * A native method that is implemented by the 'vadjni' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();

    public static native long createInstance(String config_path);

    public static native int destroyInstance(long instance);

    public static native int reset(long instance);

    public static native int chunkSizeSamples(long instance);

    public static native int feedForward(long instance, float[] chunk);
}

File: runtime/examples/audio_classification/android_demo/app/src/main/java/com/example/cls/MainActivity.kt
package com.baidu.paddlespeech.cls

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    private lateinit var tvContent: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tvContent = findViewById(R.id.tv_content)

        nClsCreateInstance()
        tvContent.text = nClsFeedForward()
        nClsReset()
        nClsDestroyInstance()
    }

    external fun nClsCreateInstance(): Boolean
    external fun nClsDestroyInstance(): Boolean
    external fun nClsFeedForward(): String
    external fun nClsReset(): Boolean

    companion object{
        init {
            System.loadLibrary("native-lib")
        }
    }
}